clients/admin-ui/src/features/system/VendorSelector.tsx

```diff
@@ -130,165 +134,185 @@ const VendorSelector = ({
     opt.label.toLowerCase().startsWith(searchParam.toLowerCase())
   );
 
+  const isTypeahead = !field.value && !values.vendor_id;
   const hasVendorSuggestions = !!searchParam && suggestions.length > 0;
 
+  const handleClear = () => {
+    setValue("");
+    setSearchParam("");
+    setFieldValue("vendor_id", undefined);
+    onVendorSelected(undefined);
+  };
+
   const handleTabPressed = () => {
+    if (!searchParam) {
+      return;
+    }
     if (suggestions.length > 0 && searchParam !== suggestions[0].label) {
       setSearchParam(suggestions[0].label);
       setValue(suggestions[0].value);
     }
   };
 
-  const handleChange = (
+  const handleSelectChange = (
     newValue: SingleValue<Option>,
     actionMeta: ActionMeta<Option>
   ) => {
+    if (actionMeta.action === "clear") {
+      handleClear();
+      return;
+    }
     setValue(newValue ? newValue.label : "");
-    setTouched({ ...touched, vendor_id: true, name: !fieldsSeparated });
+    setTouched(
+      { ...touched, vendor_id: true, name: true },
+      // do not validate if a new option was created; this prevents
+      // incorrectly showing a "required field" error while a value is in
+      // the field
```

Comment:
The only scenario I can think of where not validating here (and the similar call below this) would cause problems would be when manually creating a system with a taken name, but validation for taken name seems to work anyway?  I tried also to use Formik's manual `validateField` methods for this but couldn't make them work, but this seems okay instead.